### DISCUSSION TITLE
push local clone with LFS object to new repository, preserving history?

### DISCUSSION THREAD

**Discussion by ukuko** [git-lfs/git-lfs#4690] [Q&A] [answered]
Hello, 
hopefully there is any solution to my issue:

we have a local clone which include LFS objects.

we had issues with the repository server (repository no longer reachable) and we want to push this clone to a new fresh repository, preserving history and the LFs objects.

is this possible at all?

a post in SO suggests to use first:  

git lfs fetch --all

but this will not work as -as mentioned- repository is not reachable:

https://stackoverflow.com/questions/49349974/move-git-directory-from-one-lfs-enabled-repo-to-another-also-lfs/49366471#49366471

any idea?

**Comment by bk2204** [comment, marked as answer]
Hey,

The way Git LFS works is that it doesn't store all of the data in the local repository by default.  That's because many of the files are large, and the goal is to avoid needing to hold those objects in every clone.  So if you can't access the server and nobody has all of the objects, then unfortunately you've probably lost data.  You may want to see what data recovery you can do on the repository server, if it's possible.

If you just want to push what you have and move on, then there's the config option `lfs.allowincompletepush` which will permit you to push incomplete data, provided your new server will let you.

**Reply by ukuko** [reply]
Hi Brian, 
thanks for your answer. one question more, if I use lfs.allowincompletepush I will get history just without LFS objects? but at least I will get current head objects?

**Reply by bk2204** [reply]
You will get the history with whatever LFS objects you have on your local system and no more.  The pointer files will remain in the Git history, but for any of those pointer files that don't have LFS files locally, the LFS contents will be inaccessible.

**Reply by ukuko** [reply]
Ok, thanks
